server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -422,6 +426,13 @@
                         "Indicates whether the host in down state can be put into maintenance state so thats its not enabled after it comes back.",
                         true, ConfigKey.Scope.Zone, null);
 
+    protected final ConfigKey<Integer> KvmVmMigrateSpeed = new ConfigKey<>("Advanced", Integer.class, KVM_VM_MIGRATE_SPEED, "-1",
+            "set the vm migrate speed (in MiB/s) on KVM. By default, it will try to guess the speed of the guest network (in MBps).", true);
+    protected final ConfigKey<Integer> KvmVmMigrateDowntime = new ConfigKey<>("Advanced", Integer.class, KVM_VM_MIGRATE_DOWNTIME, "-1",
+            "Sets maximum tolerable time in milliseconds for which the domain is allowed to be paused at the end of live migration on KVM.", true);
+    protected final ConfigKey<Integer> KvmVmMigratePauseAfter = new ConfigKey<>("Advanced", Integer.class, KVM_VM_MIGRATE_PAUSE_AFTER, "-1",
+            "Set an upper limit in milliseconds for how long live migration on KVM should wait, at which point VM is paused and migration will finish quickly.  Less than 1 means disabled.", true);
+
```

Comment:
these belong in the hypervisor plugin for KVM, not here.

Reply:
@DaanHoogland do you know the right place where can I add it? I didnt find any file for kvm specific global settings.

Reply:
I'd say in the libvirt guru or - resource, I'd say. But to tell you the truth, I really don't know. As they are used in the AgentManager, maybe there. But for sure these shouldn't be initialised in `ConfigurationManagerImpl.initMessageBusListener()`.
For vmware there are such `ConfigKey`'s stored in the `VmwareGuru` and in a `VmwareManager`. Both seem resonable. In the KVM case the `KVMGuru` (which actually should be in the kvm hypervisor plugin, soit).

Reply:
> I'd say in the libvirt guru or - resource, I'd say. But to tell you the truth, I really don't know. As they are used in the AgentManager, maybe there. But for sure these shouldn't be initialised in `ConfigurationManagerImpl.initMessageBusListener()`.
> For vmware there are such `ConfigKey`'s stored in the `VmwareGuru` and in a `VmwareManager`. Both seem resonable. In the KVM case the `KVMGuru` (which actually should be in the kvm hypervisor plugin, soit).

agree with @DaanHoogland 
`KVMGuru` looks like better place.

Reply:
@DaanHoogland @weizhouapache `KVmguru` doesnt have any code related to configKeys. shall i add that also?

Reply:
Yes, makes sense to me @ravening

Reply:
@DaanHoogland @weizhouapache moved the settings to `KVMguru`